state_machine/applications/flight/Tasks/telemetry.py

```diff
@@ -20,15 +19,17 @@ async def main_task(self):
         try:
             self.write_beacon()
         except Exception:
-            files.mkdirp('/sd/logs/beacon/')
+            files.mkdirp('/sd/logs/telemetry/')
             self.write_beacon()
 
     def write_beacon(self):
-        currTime = time.time()
-        TIMEINTERVAL = 1000
-        log_directory = "/sd/logs/"
-        current_file = f"{log_directory}beacon/{int(currTime//TIMEINTERVAL)}.txt"
+        t = cubesat.rtc.datetime
+        hour_stamp = f'{t.tm_year}.{t.tm_mon}.{t.tm_mday}.{t.tm_hour}'
+        current_file = f"/sd/logs/telemetry/{hour_stamp}.txt"
+        time_packet = logs.time_packet(t)
         beacon_packet = logs.beacon_packet()
+        system_packet = logs.system_packet()
         file = open(current_file, "ab+")
-        file.write(bytearray(beacon_packet))
+        file.write(bytearray(time_packet) + bytearray(beacon_packet) + bytearray(system_packet))
+        file.write(bytearray("\n", 'utf-8'))
```

Comment:
Since we are writing the file as binary, we won't be able to know if a `\n` is part of the packet, or if it is data (without looking at the length of the telemetry packet), so I'd say we leave out the `\n` and we will parse these files by reading `len(telemetry_packet)` bytes at a time.